src/memray/commands/common.py

```diff
@@ -127,9 +127,13 @@ def write_report(
         merge_threads: Optional[bool] = None,
         inverted: Optional[bool] = None,
         temporal: bool = False,
+        max_memory_records: int = 0,
     ) -> None:
         try:
-            reader = FileReader(os.fspath(result_path), report_progress=True)
+            kwargs = {}
+            if max_memory_records > 0:
+                kwargs["max_memory_records"] = max_memory_records
```

Comment:
Huh - so you're making it impossible to set the max to 0, even though the parameter defaults to 0? That seems... weird. I'd have expected 0 to mean "don't include any memory records", and I'd have expected a different sentinel for "unlimited" (I'd use `None`, because `max_memory_records=None` does seem to read as "no limit" to me, but I could live with `-1` as a sentinel if you don't want to make this `Optional`)

Reply:
Yeah I agree, let's mark this as Optional as use None to mark the default

Reply:
Also, note the default is set to a very big number, not "No limit"